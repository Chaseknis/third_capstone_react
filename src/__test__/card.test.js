import React from 'react';
import { render, screen } from '@testing-library/react';
import Card from '../routers/card';

jest.mock('../redux/api/api', () => ({
  __esModule: true,
  default: jest.fn(() => ({
    name: 'New York',
    country: 'US',
    temp: 18.5,
    iconURL: 'https://www.example.com/image.png',
    description: 'Clear Sky',
  })),
}));

describe('Card', () => {
  test('renders card details', async () => {
    render(<Card city="New York" />);
    const cityName = await screen.findByTestId('city-name');
    const temp = await screen.findByText(/18\.5Â°C/);
    const description = await screen.findByText(/Clear Sky/);
    const image = await screen.findByAltText(/weather_icon/);
    const arrow = await screen.findByTitle(/arrow/);
    expect(cityName).toBeInTheDocument();
    expect(temp).toBeInTheDocument();
    expect(description).toBeInTheDocument();
    expect(image).toBeInTheDocument();
    expect(arrow).toBeInTheDocument();
    expect(cityName).toHaveTextContent('New York, US');
  });
});
