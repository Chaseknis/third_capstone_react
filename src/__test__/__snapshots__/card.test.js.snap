// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card component 1`] = `
<div
  className="card_details_wrapper"
/>
`;
